plugins/woocommerce/src/Admin/RemoteInboxNotifications/RemoteInboxNotificationsEngine.php

```diff
@@ -233,4 +245,79 @@ public static function get_note_from_db( $note_from_db ) {
 
 		return $note_from_db;
 	}
+
+	/**
+	 * Add the debug tools to the WooCommerce debug tools (WooCommerce > Status > Tools).
+	 *
+	 * @param array $tools a list of tools.
+	 *
+	 * @return mixed
+	 */
+	public static function add_debug_tools( $tools ) {
+		$tools['refres_remote_inbox_notifications'] = array(
+			'name'     => __( 'Refresh Remote Inbox Notifications', 'woocommerce' ),
+			'button'   => __( 'Refresh', 'woocommerce' ),
+			'desc'     => __( 'This will refresh the remote inbox notifications', 'woocommerce' ),
+			'callback' => function () {
+				RemoteInboxNotificationsDataSourcePoller::get_instance()->read_specs_from_data_sources();
```

Comment:
Since this would call an external API, I think we might need to check the user's consent first. If it's not consented, we might not be allowed to run it 🤔 

An easy way might be to only enable the tool if it's checked.

Another alternative might be to:
1. Append to the description saying they should check the checkbox in `WooCommerce > Settings > Advanced > WooCommerce.com > Display suggestions within WooCommerce`
2. Disable the `Refresh` button unless it's checked, similar to the check [here](https://github.com/woocommerce/woocommerce/blob/d54bbc05ffbf2f752bb0c6b52115051d53e3137b/plugins/woocommerce/src/Internal/Admin/Events.php#L232-L245)

Reply:
Good catch! Let's hide these tools if user hasn't agreed.

Reply:
@ilyasfoo  Fixed in [Do not attempt to delete a note if it is not found](https://github.com/woocommerce/woocommerce/pull/48961/commits/4457d67bdf6c49609f96d10ac327f0697141d152)

Reply:
@ilyasfoo Fixed in [Do not add the tools if tracking is disabled](https://github.com/woocommerce/woocommerce/pull/48961/commits/d06a1518569eb41b14c5552bf7985460ef5b5ad7)

Reply:
Thanks, @moon0326! Sorry I was vague, I think we should follow [this logic](https://github.com/woocommerce/woocommerce/blob/97656a907decc39a45f82f5a65356816b97c1ebd/plugins/woocommerce/src/Internal/Admin/Events.php#L232-L245) instead.

Reply:
@moon0326 What do you think? I think we should follow the same logic to ensure consistency

Reply:
@ilyasfoo Good idea! Updated in https://github.com/woocommerce/woocommerce/pull/48961/commits/226aab04992915477539627f74ad37845d72b5be

Reply:
Looks good, thanks!